execution/src/execution.rs

```diff
@@ -262,6 +262,45 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
 
         Ok(Some(tx))
     }
+
+    pub async fn get_logs(
+        &self,
+        filter: &Filter,
+        payloads: &BTreeMap<u64, ExecutionPayload>,
+    ) -> Result<Vec<Log>> {
+        let logs = self.rpc.get_logs(filter).await?;
```

Comment:
So right now proving a single receipt's existence is very slow. We are working on a potential change to the Ethereum RPC to make it much more efficient, but for now we just have to live with it. A concern I have is that users will make a query for tons of logs, and then Helios will be stuck processing this request for many minutes.

I think a good solution for now would be to limit the number of logs that can be returned to say ~5 logs. Alchemy actually does this, but with a max of 10k logs instead.

The question is whether we should simply truncate the logs or throw an error. I'm leaning towards throwing an error since I don't want Helios to ever return incomplete data.

Reply:
> The question is whether we should simply truncate the logs or throw an error. I'm leaning towards throwing an error since I don't want Helios to ever return incomplete data.

totally agree